[ticket/13891] Added reparser:list and reparser:reparse to CLI

s9e opened the pull request:
https://tracker.phpbb.com/browse/PHPBB3-13891

- [x] Depends on https://tracker.phpbb.com/browse/PHPBB3-13803 / #3579

Nicofuma commented on phpBB/phpbb/console/command/reparser/list_all.php:
```diff
@@ -0,0 +1,70 @@
+	{
+		parent::__construct($user);
+		$this->reparser_names = array();
+		foreach (array_keys($container->findTaggedServiceIds('text_reparser.plugin')) as $name)
```
use a \phpbb\di\service_collection instead

Nicofuma commented:
beside my comment, +1

s9e commented:
@Nicofuma I replaced the collection, added a progress bar and added a couple of options in case someone wants to test it on a smaller range of posts. AFAIK it should be good to go.

```bash
./phpBB/bin/phpbbcli.php reparser:reparse post_text \
	-vvv              \
	--ansi            \
	--range-min  123  \
	--range-max  4567 \
	--range-size 50
```

Nicofuma commented:
looks great, just for the design I think it could be great to use the symfony style

for the style: https://github.com/phpbb/phpbb/pull/3514/files#diff-2b8f0912fa7a05395a2997d0e1949a25R98

for the progress bar: https://github.com/phpbb/phpbb/pull/3514/files#diff-5f4a6cdd29f5a7c27bdd0b98356cfef4R190 ($this->io is an instance of a Symfony Style)

and with a nice success/error message at the end: https://github.com/phpbb/phpbb/pull/3514/files#diff-5f4a6cdd29f5a7c27bdd0b98356cfef4R175

s9e commented:
@Nicofuma I don't know when I'll be able to look into that. I'm busy investigating memory usage/chasing down possible memory leaks.

@CHItA Is there a way for you to move the CLI styling to the `phpbb\console` namespace and make it reusable?

Nicofuma commented:
Actually the CI styling is already in Symfony. In this file there is just a few tweaks for the installer.

I'll send you a PR to improve the design.

s9e commented:
Awesome, thanks! :+1:

s9e commented:
!set WIP

Nicofuma commented:
why WIP?

s9e commented:
@Nicofuma How do you feel about showing the elapsed/estimated time and memory usage in the progress bar?

![](https://area51.phpbb.com/phpBB/download/file.php?id=3896&mode=view)

s9e commented:
!unset WIP

Nicofuma commented:
I like it but why there is 2 times?
